src/scenic/syntax/ast.py

```diff
@@ -1,20 +1,187 @@
 import ast
+from typing import Optional, Union
+
 
 class AST(ast.AST):
     "Scenic AST base class"
+
     def __init__(self, *args: any, **kwargs: any) -> None:
         super().__init__(*args, **kwargs)
 
+
 # special statements
 class EgoAssign(AST):
+    __match_args__ = ("value",)
+
     def __init__(self, value: any, *args: any, **kwargs: any) -> None:
         super().__init__(*args, **kwargs)
         self.value = value
         self._fields = ["value"]
 
+
+# Instance Creation
+
+
 class New(AST):
-    def __init__(self, className: str, specifiers: list, *args: any, **kwargs: any) -> None:
+    __match_args__ = ("className", "specifiers")
+
+    def __init__(
+        self, className: str, specifiers: list, *args: any, **kwargs: any
+    ) -> None:
         super().__init__(*args, **kwargs)
         self.className = className
         self.specifiers = specifiers
         self._fields = ["className", "specifiers"]
+
+
+# Specifiers
+
+
+class WithSpecifier(AST):
+    __match_args__ = ("prop", "value")
+
+    def __init__(self, prop: str, value: ast.AST, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+        self.prop = prop
+        self.value = value
+        self._fields = ["prop", "value"]
+
+
+class AtSpecifier(AST):
+    __match_args__ = ("position",)
+
+    def __init__(self, position: ast.AST, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+        self.position = position
+        self._fields = ["position"]
+
+
+class OffsetBySpecifier(AST):
+    __match_args__ = ("offset",)
+
+    def __init__(self, offset: ast.AST, *args: any, **kwargs: any) -> None:
+        super().__init__(*args, **kwargs)
+        self.offset = offset
+        self._fields = ["offset"]
+
+
+class OffsetAlongSpecifier(AST):
+    __match_args__ = ("direction", "offset")
+
+    def __init__(
+        self, direction: ast.AST, offset: ast.AST, *args: any, **kwargs: any
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.direction = direction
+        self.offset = offset
+        self._fields = ["direction", "offset"]
+
+
+class PositionSpecifier(AST):
```

Comment:
Let's give this a more precise name, since there are other "position" specifiers. Maybe `DirectionOfSpecifier`?

Reply:
Renamed it to `DirectionOfSpecifier` e69e2e7427200ae56b203cb6d982786b0ecf67a5